monkey/tests/unit_tests/infection_monkey/network/test_ping.py

```diff
@@ -0,0 +1,175 @@
+import subprocess
+from unittest.mock import MagicMock
+
+import pytest
+
+from infection_monkey.network import ping
+
+LINUX_SUCCESS_OUTPUT = """
+PING 192.168.1.1 (192.168.1.1) 56(84) bytes of data.
+64 bytes from 192.168.1.1: icmp_seq=1 ttl=64 time=0.057 ms
+
+--- 192.168.1.1 ping statistics ---
+1 packets transmitted, 1 received, 0% packet loss, time 0ms
+rtt min/avg/max/mdev = 0.057/0.057/0.057/0.000 ms
+"""
+
+LINUX_NO_RESPONSE_OUTPUT = """
+PING test-fake-domain.com (127.0.0.1) 56(84) bytes of data.
+
+--- test-fake-domain.com ping statistics ---
+1 packets transmitted, 0 received, 100% packet loss, time 0ms
+"""
+
+WINDOWS_SUCCESS_OUTPUT = """
+Pinging 10.0.0.1 with 32 bytes of data:
+Reply from 10.0.0.1: bytes=32 time=2ms TTL=127
+
+Ping statistics for 10.0.0.1:
+    Packets: Sent = 1, Received = 1, Lost = 0 (0% loss),
+Approximate round trip times in milli-seconds:
+    Minimum = 2ms, Maximum = 2ms, Average = 2ms
+"""
+
+WINDOWS_NO_RESPONSE_OUTPUT = """
+Pinging 10.0.0.99 with 32 bytes of data:
+Request timed out.
+
+Ping statistics for 10.0.0.99:
+    Packets: Sent = 1, Received = 0, Lost = 1 (100% loss),
+"""
+
+MALFORMED_OUTPUT = """
+WUBBA LUBBA DUB DUBttl=1a1 time=0.201 ms
+TTL=b10
+TTL=1C
+ttl=2d2!
+"""
+
+
+@pytest.fixture
+def patch_ping_subprocess(monkeypatch):
+    def inner(mock_obj):
+        monkeypatch.setattr("subprocess.Popen", MagicMock(return_value=mock_obj))
+
+    return inner
+
+
+@pytest.fixture
+def patch_ping_subprocess_with_ping_output(patch_ping_subprocess):
+    def inner(ping_output):
+        mock_ping = MagicMock()
+        mock_ping.communicate = MagicMock(return_value=(ping_output, ""))
+        patch_ping_subprocess(mock_ping)
+
+    return inner
+
+
+@pytest.fixture
+def patch_ping_subprocess_raise_timeout_expired(patch_ping_subprocess):
+    mock_ping = MagicMock()
+    mock_ping.communicate = MagicMock(side_effect=subprocess.TimeoutExpired(["test-ping"], 10))
+
+    patch_ping_subprocess(mock_ping)
+
+
+@pytest.fixture
+def set_os_linux(monkeypatch):
+    monkeypatch.setattr("sys.platform", "linux")
+
+
+@pytest.fixture
+def set_os_windows(monkeypatch):
+    monkeypatch.setattr("sys.platform", "win32")
+
+
+@pytest.mark.usefixtures("set_os_linux")
+def test_linux_ping_success(patch_ping_subprocess_with_ping_output):
```

Comment:
I'm guessing for readability because of the "_Sometimes test functions do not directly need access to a fixture object._" [here](https://docs.pytest.org/en/6.2.x/fixture.html#use-fixtures-in-classes-and-modules-with-usefixtures)?